integration/spark/src/test/java/io/openlineage/spark/agent/SparkContainerIntegrationTest.java

```diff
@@ -206,6 +226,22 @@ public void testPysparkRddToTable() throws IOException, InterruptedException {
             .withBody(json(completeTableEvent, MatchType.ONLY_MATCHING_FIELDS)));
   }
 
+  @Test
+  public void testPysparkKafkaReadWrite() throws IOException {
+    kafka = makeKafkaContainer();
+    kafka.start();
+
+    pyspark =
+        makePysparkContainerWithDefaultConf(
+            "testPysparkKafkaReadWriteTest",
+            "--packages",
+            "org.apache.spark:spark-sql-kafka-0-10_2.11:" + System.getProperty("spark.version"),
+            "/opt/spark_scripts/spark_kafka.py");
+
+    pyspark.setWaitStrategy(Wait.forLogMessage(".*ShutdownHookManager: Shutdown hook called.*", 1));
+    pyspark.start();
```

Comment:
Possibly add a comment here:
```
//TODO add lineage validation test
```
We might want to discuss how to add this feature incrementally.

Reply:
@julienledem i would suggest following approach:
Before supporting spark streaming capturing what is a whole new feature. I would start with capturing lineage from various data sources as kafka, delta, hive tables, csv, ... on simple batch read/write. 

For example fixing writing to kafka : https://github.com/OpenLineage/OpenLineage/issues/280

We also consider in future switching from spark to flink with beam so also provide support those. 
What are your thoughts or approach on adding features into openlineage?

Reply:
I think that makes sense. We can easily review adding coverage for each source independently to the current batch processing coverage. For streaming, I'm making a proposal on the project slack. Please come chime in.

Reply:
So basically:
1. increasing batch datasource coverage should go as PRs to main 1 by 1. 
2. For the streaming implementation: My proposal is to start a branch and incrementally write PRs against it until we are ready to merge it in main